### DISCUSSION TITLE
suggestion: support latex math expressions using remark-math and rehype-katex

### DISCUSSION THREAD

**Discussion by gyunseo** [satnaing/astro-paper#347] [General]
I think support latex math expression in markdown is a good option like the following image.
![image](https://github.com/satnaing/astro-paper/assets/48322716/260bd039-ba30-4833-99da-4551ac0ac1ee)

Thank you
Best Regards
@gyunseo

**Comment by satnaing** [comment]
Appreciate your suggestion and effort.
However, I'm thinking of adding a `how-to` blog post for adding support latex math expressions. This is because AstroPaper is aimed to be as minimal as possible; and, besides, not everyone using AstroPaper would need this feature. But if someone does, they can opt in this feature by following the `how-to` blog post guide.

**Comment by andreyyao** [comment]
I tried adding katex using the same process you described, and it worked quite well, except that the colors are messed up in dark mode:
![image](https://github.com/satnaing/astro-paper/assets/96220436/51bbfe81-4ae8-4417-9c91-9e0ff891d1a0)

As you can see, the math text is very very faint. Does anyone have any idea how to fix this?

**Comment by caleb-shelton** [comment]
I followed this tutorial and didn't have this issue @andreyyao https://blog.alexafazio.dev/blog/render-latex-in-astro/

**Comment by andreyyao** [comment]
@caleb-shelton I just did a fresh install, and it's the same issue. Which version of remark-math and rehype-katex are you using?

EDIT: Upon further inspection, I think tailwindcss or the dark mode toggle is not playing right with the katex expressions. I got a temporary fix working by overwriting this css in layouts/PostDetails.astro:
```
 /* This is fixing the dim color of katex in dark mode */
 .prose {
     color: var(--color-text-base);
 }
```

**Comment by gyunseo** [comment]
@andreyyao I didn't know that issue since I always use light mode. I've found the same issue with yours on my blog. I fixed it using your modification on `layouts/PostDetails.astro`. thanks so much! :D

**Comment by satnaing** [comment]
Convert this issue into a discussion for future reference. 
Thank you to everyone who participated in the conversation.
